File: client/scalaz/src/main/scala/org/openapitools/client/api/SdtmigProduct.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import SdtmigProduct._

case class SdtmigProduct (
  name: Option[String],
label: Option[String],
description: Option[String],
source: Option[String],
effectiveDate: Option[String],
registrationStatus: Option[String],
version: Option[String],
links: Option[SdtmigProductLinks],
classes: Option[List[SdtmigClass]])

object SdtmigProduct {
  import DateTimeCodecs._

  implicit val SdtmigProductCodecJson: CodecJson[SdtmigProduct] = CodecJson.derive[SdtmigProduct]
  implicit val SdtmigProductDecoder: EntityDecoder[SdtmigProduct] = jsonOf[SdtmigProduct]
  implicit val SdtmigProductEncoder: EntityEncoder[SdtmigProduct] = jsonEncoderOf[SdtmigProduct]
}

File: src/cdisc_library_api/specs/sdtmig_product.clj
(ns cdisc-library-api.specs.sdtmig-product
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [cdisc-library-api.specs.sdtmig-product-links :refer :all]
            [cdisc-library-api.specs.sdtmig-class :refer :all]
            )
  (:import (java.io File)))


(def sdtmig-product-data
  {
   (ds/opt :name) string?
   (ds/opt :label) string?
   (ds/opt :description) string?
   (ds/opt :source) string?
   (ds/opt :effectiveDate) string?
   (ds/opt :registrationStatus) string?
   (ds/opt :version) string?
   (ds/opt :_links) sdtmig-product-links-spec
   (ds/opt :classes) (s/coll-of sdtmig-class-spec)
   })

(def sdtmig-product-spec
  (ds/spec
    {:name ::sdtmig-product
     :spec sdtmig-product-data}))
